auraed/src/runtime/cell_service/executables/executable.rs

```diff
@@ -31,7 +42,16 @@ enum ExecutableState {
 impl Executable {
     pub fn new<T: Into<ExecutableSpec>>(spec: T) -> Self {
         let ExecutableSpec { name, description, command } = spec.into();
-        Self { name, description, state: ExecutableState::Init { command } }
+        let stdout = Arc::new(LogChannel::new(format!("{}::stdout", name)));
+        let stderr = Arc::new(LogChannel::new(format!("{}::stderr", name)));
```

Comment:
I'm wondering why these are `Arc` wrapped. I haven't tried changing anything to see, but I feel like they shouldn't be.

LogChannel in general has a weird interface that we should fix.

Reply:
so i can pass them into the thread (which is currently commented out).

Reply:
LogChannel should be clone-able (if we add `Clone` to the derive). `Sender<LogItem>` is expected to be cloned and I assume `name` will always be small enough to be cheap to clone.

We can refactor later though.

Reply:
yeah .. i might need an ExecutableInner anyway to allow the Process to be shared with the logging thread too. so i think we'll need 

```
struct Executable {
  inner: Arc<Mutex<ExecutableInner>>;
};

struct Executable {
  process: Process;
  stdout: LogChannel;
 ...
};
```

at which point everything gets a bit more complex but also easier.

Reply:
Do we not want the same functionality on all `Process` wrappers (i.e., `Executable` & `NestedAurae`)?  In which case, I think, we would continue to expand the functionality in `Process`, and the wrappers would proxy to it, like with `kill` and now `read_stdout/err`

Reply:
i thought Process should be the low level view.. ie, exposing information but not doing anything with it. then Executable (and presumably NestedAurae) can decide what to do with that data.  so Executable will push it to Observe (and optionally trace) whereas NestedAurae would only trace it.

Reply:
i went with the `ExecutableInner` and it's much cleaner.